### DISCUSSION TITLE
[Feature Request] Opera Mini Web Browser in Native Apps

### DISCUSSION THREAD

**Discussion by vaucci** [OnionUI/Onion#1466] [Ideas]
Hey guys, is everything ok?  I would like to leave a suggestion for Opera Mini (java version) that ran on old cell phones as a native Web Browser for Onion OS, in addition to being lightweight, it has a built-in cursor and keyboard, ideal for browsing rom sites or forums, here Here is the link to the available java versions:

[m.opera.com](http://m.opera.com/?act=opts&rnd=1965170916&vid=0xec0b4ef9eef29805&ua=Nokia100%2F2.0+%282.0423.0_rc02%29+SymbianOS%2F7.0s+Series60%2F2.1+Profile%2FMIDP-2.0+Configuration%2FCLDC-1.0)

**Comment by XK9274** [comment]
Hey,

So I've tried to get java cores working previously (FreeJ2ME).

Compiled the core aswell as getting my hands on a JDK that works on this arch. The good news is it boots, the bad news is it's in a very poor state (slow, fonts missing, buggy). Not only does java use the bulk of the CPU, the emu/core also needs these resources which the miyoo only has 2 cores and 128mb ram.

The JDK is 400mb, it may be that we only needs parts of it but, I don't think I'll be spending any more time on it ATM.

As for browsers, does the Miyoo really NEED a browser? They're heavy on resources in most cases and navigation will be difficult, security will be poor, the experience will be less than ideal. I'm a big fan of doing things just because but, in my opinion very few will pursue this, creating it is one thing but maintaining it and keeping up with what people will expect of a web browser is a whole other time-sink. 

It's likely not gonna happen unless there's big leaps in java emu, the best candidate is probably SquirrelJME which you can check out here: https://multiphasicapps.net/doc/ckout/readme.mkd

With that said, as always if anyone comes across this post and wants to take over from where I got to ill hand over the files. I'll post them in Onions Dev chat, who knows, someone else might have a go.

**Comment by vaucci** [comment]
Hello friend, thanks for the answer, I partly agree with you that everything needed to play, Miyoo already offers, but sometimes we need a break between one game and another and venture into mobile version web pages like Opera Mini, it would be Quite an adventure, that's why I suggested it here, sometimes I don't want to go back to the 400ppi, 6-inch screen on my smartphone, but rather continue playing on the Miyoo Mini Plus, even with 128mb of RAM, I believe I could access some websites reading, just as they were accessible in the days before Symbian.

You talked about the 400mb SDK you used, could you make it available for me to test? I managed to advance a little, but the SDK I used was from the RX35XX platform, so I believe it didn't work, if you were able to make it available, I would be grateful.

![miyoo-mini-plus-running-opera-mini](https://github.com/OnionUI/Onion/assets/57840299/5e745834-b548-4966-9b55-83cb10f2f9f5)
